data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/query/BindableParametersStoredQuery.java

```diff
@@ -90,18 +90,20 @@ interface Binder {
         /**
          * Bind the value.
          *
-         * @param binding The binding
+         * @param binding  The binding
+         * @param property The property that binding is done for
          * @param value    The value
          */
```

Comment:
Why is it a property and not just annotation metadata? For a custom query there are not properties. Please add a test case

Reply:
Right, because we get json type info from property (when saving an entity) or from parameter. And I wanted to have RuntimePersistentProperty.getJsonType to read once from annotation. Maybe we could use annotation metadata, will have another look.

Reply:
My point is that custom queries with parameter `xyz` might not going to have the appropriate `RuntimePersistentProperty`, so you need to use `AnnotationMetadata`. Please create a test with a table having one column not included in the entity and try to modify using a custom query.

Reply:
I see, will have to create test with manually creating schema or something.
I thought for method like this (or with unmapped property)
```
@Query("UPDATE json_entity SET json_blob = :jsonBlob WHERE id = :id")
    void updateJsonBlobById(Long id, @JsonRepresentation(type = JsonType.BLOB) SampleData jsonBlob);
```
would get annotation metadata in QueryParameterBinding from the parameter. But I will add the test and check this.

Reply:
I have this in JdbcParameterBinder for example 
```
public void bindOne(QueryParameterBinding binding, @Nullable RuntimePersistentProperty<?> property, Object value) {
            JsonDataType jsonDataType = null;
            if (binding.getDataType() == DataType.JSON) {
                jsonDataType = binding.getJsonDataType();
                if (jsonDataType == null && property != null) {
                    jsonDataType = property.getJsonDataType();
                }
            }
            setStatementParameter(ps, index, binding.getDataType(), jsonDataType, value, sqlStoredQuery);
            index++;
        }
```
and for this method using custom query (annotations not mapped in entity)
```
@Query("UPDATE patient SET appointments = :appointments WHERE name = :name")
    void updateAppointmentsByName(@Parameter String name, @TypeDef(type = DataType.JSON) List<String> appointments);
```
it finds it via parameter. I have added changes to QueryParameterBinding to read JsonRepresentationEntity from the parameter and it seems to be working.

Reply:
Yes, you were right. Parameter is not needed here, code is changed to read from metadata via QueryParameterBinding.